code/modules/mob/living/inhand_holder.dm

```diff
@@ -5,80 +5,118 @@
-	held_mob.forceMove(get_turf(held_mob))
-	held_mob.reset_perspective()
-	held_mob.setDir(SOUTH)
-	held_mob.visible_message("<span class='warning'>[held_mob] uncurls!</span>")
-	held_mob = null
-	if(del_on_release && !destroying)
-		qdel(src)
-	return TRUE
+	if(held_mob)
+		var/mob/living/m = held_mob
+		m.forceMove(get_turf(m))
+		m.reset_perspective()
+		m.setDir(SOUTH)
+		held_mob = null
+	qdel(src)
 
 /obj/item/clothing/head/mob_holder/relaymove(mob/user)
-	release()
+	to_chat(user, "<span class='warning'>You must resist in order to escape!</span>")
+	return
 
 /obj/item/clothing/head/mob_holder/container_resist()
+	if(isliving(loc))
+		var/mob/living/L = loc
+		visible_message("<span class='warning'>[src] escapes [L]!</span>")
 	release()
 
-/obj/item/clothing/head/mob_holder/drone/deposit(mob/living/L)
-	. = ..()
-	if(!isdrone(L))
-		qdel(src)
-	name = "drone (hiding)"
-	desc = "This drone is scared and has curled up into a ball!"
-
-/obj/item/clothing/head/mob_holder/drone/update_visuals(mob/living/L)
-	var/mob/living/simple_animal/drone/D = L
-	if(!D)
-		return ..()
-	icon = 'icons/mob/drone.dmi'
-	icon_state = "[D.visualAppearence]_hat"
+/mob/living/proc/mob_pickup(mob/living/L)
+	var/obj/item/clothing/head/mob_holder/holder = generate_mob_holder()
+	if(!holder)
+		return
+	drop_all_held_items()
+	L.put_in_hands(holder)
+	return
+
+/mob/living/proc/mob_try_pickup(mob/living/user)
+	if(!ishuman(user) || !src.Adjacent(user) || user.incapacitated() || !can_be_held)
+		return FALSE
+	if(user.get_active_held_item())
+		to_chat(user, "<span class='warning'>Your hands are full!</span>")
+		return FALSE
+	if(buckled)
+		to_chat(user, "<span class='warning'>[src] is buckled to something!</span>")
+		return FALSE
+	if(src == user)
+		to_chat(user, "<span class='warning'>You can't pick yourself up.</span>")
+		return FALSE
+	visible_message("<span class='warning'>[user] starts picking up [src].</span>", \
+					"<span class='userdanger'>[user] starts picking you up!</span>")
+	if(!do_after(user, 20, target = src))
+		return FALSE
+
+	if(user.get_active_held_item()||buckled)
+		return FALSE
+
+	visible_message("<span class='warning'>[user] picks up [src]!</span>", \
+					"<span class='userdanger'>[user] picks you up!</span>")
+	to_chat(user, "<span class='notice'>You pick [src] up.</span>")
+	mob_pickup(user)
+	return TRUE
+
+/mob/living/AltClick(mob/user)
+	mob_try_pickup(user)
+	..()
+
+
+// I didn't define these for mobs, because you shouldn't be able to breathe out of mobs and using their loc isn't always the logical thing to do.
+/obj/item/clothing/head/mob_holder/return_air()
+	var/atom/location = loc
+	if(location)
+		return location.loc.return_air(args)
+
+/obj/item/clothing/head/mob_holder/assume_air(datum/gas_mixture/env)
+	var/atom/location = loc
+	if(location)
+		return location.loc.assume_air(env)
+
+/obj/item/clothing/head/mob_holder/remove_air(amount)
+	var/atom/location = loc
+	if(location)
+		return location.loc.remove_air(amount)

```

Comment:
But these two have some issues.
Let's say, the holder happens to be inside a backpack worn by some mob, location.loc will be then said mob, then mob remove_air(amount) or assume_air(amount) will be called, which returns null.

A loop like this should be enough.
```
	var/atom/location = loc
	if(!loc)
		return //null
	var/turf/T = get_turf(loc)
	while(location != T)
		location = location.loc
		if(ismob(location))
			return location.loc.proc(amount) //assume_air or remove_air
	return loc.proc(amount) //assume_air or remove_air
```